app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,20 @@
+class PostsController < ApplicationController
+  # actions will go here
+  def index
+    @posts = Post.all
+  end
+
+  def show
+    puts params
+    @post = Post.find(params[:id])
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def edit
+    @post = Post.find(params[:id])
+  end
```

Comment:
You're going to need `create` and `update` controller actions if `new` and `edit` are going to serve a purpose, respectively. Guessing you're implementing this in your project but let me know if you have any questions about this.